runtime/doc/vim9.txt

```diff
@@ -1055,7 +1055,7 @@ variable, since they cannot be deleted with `:unlet`.  E.g.: >
 
 The values can also be useful as the default value for an argument: >
 	def MyFunc(b: blob = null_blob)
-	   if b == null_blob
+	   if b == null
```

Comment:
is that correct? According to the descriptions in `null-variables` , it should be an empty container, so `null_blob`? I would also think for clarity, the value should be compared against `null_blob`, since that's what we use as default argument explicitly

Reply:
> is that correct? According to the descriptions in `null-variables` , it should be an empty container, so `null_blob`? I would also think for clarity, the value should be compared against `null_blob`, since that's what we use as default argument explicitly

Take a look at the example at |null-compare|. The description of it says
```
For familiar null compare semantics, do not use null_<type> in a comparison,
then it easy to distinguish an empty container from a null container:
```
You may be assuming that equality is transitive.

More specifically to the exmple you bring up run, if I run
```
vim9script

def F(b: blob = null_blob)
    if b == null_blob
        echo "NO ARG"
    else
        echo "detected arg:" b
    endif
    echo "using blob:" b
enddef
var b0 = 0z
F()
echo ' '
F(b0)
```
The output is
```
NO ARG
using blob: 0z

NO ARG
using blob: 0z
```
Is that what you would expect? It looks very wrong to me.

Changing the compare to `b == null`, produces the output
```
NO ARG
using blob: 0z

detected arg: 0z
using blob: 0z
```

Maybe I should change the description at the example you're discussing.
But change it how? Maybe add "see |null-compare|" at a minimum?